xapian-core/include/xapian/mset.h

```diff
@@ -74,6 +80,36 @@ class XAPIAN_VISIBILITY_DEFAULT MSet {
     /// Destructor.
     ~MSet();
 
+    /** Assigns new weights and updates MSet.
+     *  Dereferencing the Iterator should return a double.
+     *  Iterator should dereference weights in the original order of documents.
+     *  Weights are assigned as per the original order of documents.
+     *
+     *  @param begin	Begin iterator.
+     *  @param end	End iterator.
+     *
+     *  @exception Xapian::InvalidArgument is thrown if the total number of
+     *  elements in the input dont match the total number of documents in MSet.
+     *
```

Comment:
"dont" is missing an apostrophe ("don't"), but actually it should be "doesn't" (it's the number doesn't match, so singular).

Extra content-less line at end of comment block.